adafruit_vc0706.py

```diff
@@ -193,16 +247,331 @@ def resume_video(self):
+                frame_length -= 32
+        # Return to our original directory.
+        for directory in name[:-1]:
+            chdir("..")
+        return True
+
+    def read_picture_into(self, buf):
+        """Read the next bytes of frame/picture data into the provided buffer.
+        Returns the number of bytes written to the buffer (might be less than
+        the size of the buffer).  Buffer MUST be a multiple of 4 and 100 or
+        less.  Suggested buffer size is 32.
+        """
+        bufflen = len(buf)
+        if bufflen > 256 or bufflen > (len(self._buffer) - 5):
+            raise ValueError("Buffer is too large!")
+        if bufflen % 4 != 0:
+            raise ValueError("Buffer must be a multiple of 4! Try 32.")
+        args = bytes(
+            [
+                0x0C,
+                0x0,
+                0x0A,
+                0,
+                0,
+                (self._frame_ptr >> 8) & 0xFF,
+                self._frame_ptr & 0xFF,
+                0,
+                0,
+                0,
+                bufflen & 0xFF,
+                (_CAMERA_DELAY >> 8) & 0xFF,
+                _CAMERA_DELAY & 0xFF,
+            ]
+        )
+        if not self._run_command(_READ_FBUF, args, 5, flush=False):
+            return 0
+        if self._read_response(self._buffer, bufflen + 5) == 0:
+            return 0
+        self._frame_ptr += bufflen
+        for i in range(bufflen):
+            buf[i] = self._buffer[i]
+        return bufflen
+
+    def reset(self):
+        """Resets the camera.
+        It seems that resetting after changing the image works best.
+        This also resets to the default baud rate, so re-update that too.
+        """
+        baud = self.baudrate
+        rst = self._run_command(_RESET, bytes([0x0]), 5)
+        self._uart.baudrate = 37400  # Default baud rate
+        self.set_baudrate(baud)  # Have to reset the baud rate or it fails.
+        return rst
+
+    def _run_command(self, cmd, args, resplen, flush=True):
+        if flush:
+            self._read_response(self._buffer, len(self._buffer))
+        self._send_command(cmd, args)
+        if self._read_response(self._buffer, resplen) != resplen:
+            return False
+        if not self._verify_response(cmd):
+            return False
+        return True
+
+    def _read_response(self, result, numbytes):
+        return self._uart.readinto(memoryview(result)[0:numbytes])
+
+    def _verify_response(self, cmd):
+        return (
+            self._buffer[0] == 0x76
+            and self._buffer[1] == _SERIAL
+            and self._buffer[2] == cmd & 0xFF
+            and self._buffer[3] == 0x00
+        )
+
+    def _send_command(self, cmd, args=None):
+        self._command_header[0] = 0x56
+        self._command_header[1] = _SERIAL
+        self._command_header[2] = cmd & 0xFF
+        self._uart.write(self._command_header)
+        if args:
+            self._uart.write(args)
+
+
+class VC0706Camera:
```

Comment:
The name of the device is in the library already. I think I'd simplify this library to reduce duplicating code between it and the old version.

```suggestion
class Camera:
```